apps/cmp.c

```diff
@@ -1598,12 +1602,44 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
                       opt_subject, ref_cert != NULL ? ref_cert : opt_csr);
     }
     if (opt_cmd == CMP_RR) {
-        if (opt_oldcert == NULL && opt_csr == NULL) {
-            CMP_err("missing -oldcert for certificate to be revoked and no -csr given");
-            return 0;
+        if (opt_issuer == NULL && opt_serial == NULL) {
+            if (opt_oldcert == NULL && opt_csr == NULL) {
+                CMP_err("missing -oldcert or -issuer and -serial for certificate to be revoked and no -csr given");
+                return 0;
+            }
+            if (opt_oldcert != NULL && opt_csr != NULL)
+                CMP_warn("ignoring -csr since certificate to be revoked is given");
+        } else {
+#define OSSL_CMP_RR_MSG "since -issuer and -serial is given for command 'rr'"
```

Comment:
I personally don't like `#define` in the middle of the code

Reply:
> I personally don't like `#define` in the middle of the code

I agree it looks uncommon. Yet I've checked again in https://www.openssl.org/policies/technical/coding-style.html and cannot find requirements there where to place `#define` directives.
Maybe we should add something like this there and to `util/check-format.pl`.

Reply:
Added a respective candidate entry to the "Misc stuff" list in #10725.